App.tsx

```diff
@@ -4,19 +4,164 @@ import { ImageModal } from './ImageModal';
 import { DetailsPage } from './DetailsPage';
 import { createStackNavigator } from '@react-navigation/stack';
 import { NavigationContainer } from '@react-navigation/native';
+import { createDrawerNavigator } from '@react-navigation/drawer';
+import { WeatherApp } from './WeatherApp/Weather';
+import { WeatherStackParam } from './WeatherApp/WeatherStackParam';
+import { ScannerScreen } from './BarCodeScanner/BarCodeScannerScreen';
+import { ScannerStackParam } from './BarCodeScanner/ScannerStackParam';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import FavoritesPage from './BarCodeScanner/FavoritesPage';
+import { Ionicons } from '@expo/vector-icons';
+import { ProductDetails } from './BarCodeScanner/ProductDetails';
+import Battery from './Battery/Battery';
+import { CharCreator } from './Final/CharCreator';
+import { CharViewer } from './Final/CharViewer';
+import { CharModal } from './Final/CharModal';
 
 export type StackParamList = {
   Gallery: undefined,
-  Details: { id: number; url: string};
+  Details: { id: number; url: string},
   Modal: {id: number; url: string},
 }
 
 const Stack = createStackNavigator();
+const Weather = createStackNavigator<WeatherStackParam>();
+const Scanner = createBottomTabNavigator<ScannerStackParam>();
+const Drawer = createDrawerNavigator();
```

Comment:
Each of these Navigators should have an associated type just like `Weather` and `Scanner` so that when you define your screens, your navigator can determine if they are the correct screens or not.